packages/protect/src/ffi/model-helpers.ts

```diff
@@ -421,26 +571,47 @@ export async function bulkDecryptModels<T extends Record<string, unknown>>(
   const { otherFields, operationFields, keyMap, nullFields } =
     prepareBulkModelsForOperation(models)
 
-  // Collect all fields that need to be decrypted into a single array
   const bulkDecryptPayload = operationFields.flatMap((fields, modelIndex) =>
     Object.entries(fields).map(([key, value]) => ({
       id: `${modelIndex}-${key}`,
       ciphertext: (value as EncryptedPayload)?.c ?? '',
     })),
   )
 
-  // Make a single FFI call for all fields
   const decryptedFields = await handleMultiModelBulkOperation(
     bulkDecryptPayload,
     (items) => decryptBulk(client, items),
     keyMap,
   )
 
-  // Reconstruct models
-  return models.map((_, modelIndex) => ({
-    ...otherFields[modelIndex],
-    ...nullFields[modelIndex],
-    ...Object.fromEntries(
+  // Helper function to set a nested value
+  const setNestedValue = (
```

Comment:
Same comment here on removing the duplicate `setNestedValue` functions.

Reply:
After Protect.js is a bit more streamlined - we should definitely do some refactoring exercises